nodogsplash2/files/nodogsplash.init

```diff
@@ -251,6 +251,26 @@ create_instance() {
 start_service() {
   include /lib/functions
 
+  if ! uci get firewall.nodogsplash2.path &> /dev/null ; then
+      if [ -f '/etc/nodogsplash/ndshook.sh' ] ; then
+        rm /etc/nodogsplash/ndshook.sh
+      fi
+    uci delete firewall.nodogsplash2 2> /dev/null
+    uci set firewall.nodogsplash2=include
+    uci set firewall.nodogsplash2.type=script
+    uci set firewall.nodogsplash2.path='/etc/nodogsplash/ndshook.sh'
+    uci commit firewall
+    /etc/init.d/firewall restart  2>&1 | logger
+  fi
+
+  if [ ! -f '/etc/nodogsplash/ndshook.sh' ] ; then
+    printf "#NDS Restart Hook Created by Nodogsplash. Do not modify.\n" > /etc/nodogsplash/ndshook.sh
+    printf "  if /usr/bin/ndsctl status > /dev/null; then\n" >> /etc/nodogsplash/ndshook.sh
+    printf "    echo \" * Restarting NodogSplash \"\n" >> /etc/nodogsplash/ndshook.sh
+    printf "    /etc/init.d/nodogsplash restart\n  fi\n" >> /etc/nodogsplash/ndshook.sh
```

Comment:
This might be more elegant instead all of the printf calls:
```
	cat <<-EOF > /etc/nodogsplash/ndshook.sh
		#NDS Restart Hook Created by Nodogsplash. Do not modify.
		if /usr/bin/ndsctl status > /dev/null; then
		  echo " * Restarting NodogSplash"
		  /etc/init.d/nodogsplash restart
		fi
	EOF
```

Reply:
Beauty is in the eye of the beholder!
I prefer the symmetry of the printf lines. lol

Reply:
I think it is better to put the script in a file that is installed instead of creating it programmatically. Less noise and it would be easier for the file system to compress.